src/main/java/core/basesyntax/service/handler/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.service.handler.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private static final int MIN_AMOUNT = 0;
+    private StorageDao storageDao = new StorageDaoImpl();
+
+    @Override
+    public void updateStorage(FruitTransaction transaction) {
+        if (validAmountOfFruit(transaction)) {
+            Map.Entry<Fruit, Integer> fruitAndAmount = storageDao.get(transaction.getFruitName());
+            Integer fruitAmount = fruitAndAmount.getValue();
+            Integer fruitAmountFromTransaction = transaction.getAmount();
+            fruitAndAmount.setValue(fruitAmount - fruitAmountFromTransaction);
+        }
+    }
+
+    private boolean validAmountOfFruit(FruitTransaction transaction) {
```

Comment:
Probably there is no need in `boolean` return value. We can simply make it void and just call it in `updateStorage`